platform/ios/src/MGLMapView.mm

```diff
@@ -3374,7 +3382,8 @@ - (MGLMapCamera *)cameraForCameraOptions:(const mbgl::CameraOptions &)cameraOpti
     double zoomLevel = cameraOptions.zoom ? *cameraOptions.zoom : self.zoomLevel;
     CLLocationDirection direction = cameraOptions.angle ? mbgl::util::wrap(-MGLDegreesFromRadians(*cameraOptions.angle), 0., 360.) : self.direction;
     CGFloat pitch = cameraOptions.pitch ? MGLDegreesFromRadians(*cameraOptions.pitch) : _mbglMap->getPitch();
-    CLLocationDistance altitude = MGLAltitudeForZoomLevel(zoomLevel, pitch, centerCoordinate.latitude, self.frame.size);
+    CLLocationDegrees pitchForAltitude = self.isCameraAltitudeAffectedByPitch ? pitch : 0.0;
+    CLLocationDistance altitude = MGLAltitudeForZoomLevel(zoomLevel, pitchForAltitude, centerCoordinate.latitude, self.frame.size);
```

Comment:
Pitch is by definition affected by zoom level: when the map is tilted and the zoom level changes (whether programmatically or via a gesture), the viewing distance from the focal point and the viewpoint’s distance from the ground (the altitude) both change.

These changes affect the camera getters only. So the new property is a strange way of telling the getter to effectively lie about the camera relative to what mbgl thinks the camera is. Is there no way to derive the correct value from the existing return value, perhaps by exposing MGLAltitudeForZoomLevel publicly, as proposed in #5583?

Reply:
I don't think mbgl comes in to this at all. As far as I can tell this is all happening in MGLGeometry in the iOS SDK. The pure values in mbgl are what I want.

Reply:
> The pure values in mbgl are what I want.

This is wrong.
The problem with exposing MGLAltitudeForZoomLevel is that, if you use setCamera, mbgl sets it's state from the result of MGLZoomLevelForAltitude. The stored zoomLevel is affected by pitch. I want to prevent the mbgl state from getting that value and storing the zoomLevel without regards to pitch.

Reply:
If you want to change the zoom level without changing the pitch, set a camera whose pitch is -1.

Reply:
@d-prukop and I chatted about this today. The pitch is considered twice when calculating the camera options, once for setting the camera’s actual pitch and once in calculating the zoom level. I would’ve thought that this is ideal, because the zoom level on a tilted map is really the viewing distance, but I suppose if the intention is to iteratively change the viewing distance by getting and setting the camera iteratively, then there could be a sort of feedback effect.

The essence of my concern here is that we’re adding a separate code path for a specific use case. If we really believe that the altitude-to-zoom-level conversion shouldn’t be affected by the pitch, then why not remove the pitch argument from `MGLZoomLevelForAltitude()` and `MGLAltitudeForZoomLevel()` and always assume a pitch of 0°? The obvious consequence would be that the MGLMapCamera returned by the `camera` getter would have the same `altitude` regardless of the pitch.

In other words, `isCameraAltitudeAffectedByPitch` changes the meaning of `altitude` from “shortest distance between the viewpoint and the ground” to “viewing distance from the focal point”, essentially a linear, physical-unit version of `zoomLevel`. Maybe that’s what the use case above calls for: a property for explicitly setting the zoom level of a camera instead of the altitude. We already have that in MGLMapSnapshotOptions, so why not MGLMapCamera? One reason why is that it’s awkward for an object to sport mutually exclusive properties. Any future attempt at making the `isCameraAltitudeAffectedByPitch` option public or defaulting it to `YES` will need to consider whether we instead need a `zoomLevel` property on MGLMapCamera.